env/io_posix.cc

```diff
@@ -777,8 +777,8 @@ Status PosixWritableFile::Close() {
             file_stats.st_blksize !=
         file_stats.st_blocks / (file_stats.st_blksize / 512)) {
       fprintf(stderr,
-              "WARNING: Your kernel is buggy (<= 4.0.x) and does not free"
-              "preallocated blocks on truncate. Hacking around it, but you"
+              "WARNING: Your kernel is buggy (<= 4.0.x) and does not free "
```

Comment:
Can we say "may be" or something? Who knows whether there will be such a bug in the future kernel versions.